pkgs/os-specific/linux/kernel/common-config.nix

```
@@ -490,9 +490,12 @@ let
     # of time to appear and this would hold up Linux kernel and Rust toolchain updates.
     #
     # Once Rust in the kernel has more users, we can reconsider enabling it by default.
-    rust = lib.optionalAttrs ((features.rust or false) && lib.versionAtLeast version "6.7") {
-      RUST = yes;
-      GCC_PLUGINS = no;
+    rust = lib.optionalAttrs ((features.rust or false)
+      && ((lib.versionAtLeast version "6.7" && stdenv.isx86_64)
+        || (lib.versionAtLeast version "6.9" && stdenv.isAarch64)
+        || (lib.versionAtLeast version "6.10" && stdenv.isRiscV64))) {
```

```suggestion
      && ((stdenv.isx86_64 && lib.versionAtLeast version "6.7")
        || (stdenv.isAarch64 && lib.versionAtLeast version "6.9")
        || (stdenv.isRiscV64 && lib.versionAtLeast version "6.10"))) {
```
This feels a lot more readable, especially at first glance - "archX requires at least versionY".

---

Makes sense, forced pushed.